Assets/HoloToolkit/UX/Scripts/Buttons/AnimControllerButton.cs

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+using UnityEngine;
+using System;
+using System.Collections;
+
+namespace HoloToolkit.Unity.Buttons
+{
+    /// <summary>
+    /// Anim controller button offers as simple way to link button states to animation controller parameters
+    /// </summary>
+    [RequireComponent(typeof(Animator))]
+    public class AnimControllerButton : Button
+    {
+        [Serializable]
+        public struct AnimatorControllerAction
```

Comment:
This should be in it's own file.

Reply:
Really?  You mean actually putting just the struct (that is a member of the class) in it's own file?  It's public so that the AnimActions can be populated externally.  I'll look through the rest of the code for where it's being used.

thanks,
-pat

Reply:
Yeah, it it's used outside of the `AnimControllerButton` class it should be in it's own file, otherwise we should make it private.